rhoc/pkg/cmd/deployments/snapshot/snapshot.go

```diff
@@ -0,0 +1,176 @@
+package snapshot
+
+import (
+	"errors"
+	"fmt"
+	"github.com/bf2fc6cc711aee1a0c2a/cos-tools/rhoc/pkg/service"
+	"github.com/bf2fc6cc711aee1a0c2a/cos-tools/rhoc/pkg/util/cmdutil"
+	"github.com/bf2fc6cc711aee1a0c2a/cos-tools/rhoc/pkg/util/dumper"
+	"github.com/bf2fc6cc711aee1a0c2a/cos-tools/rhoc/pkg/util/request"
+	"github.com/redhat-developer/app-services-cli/pkg/shared/factory"
+	"github.com/spf13/cobra"
+	"time"
+)
+
+const (
+	CommandName  = "snapshot"
+	CommandAlias = "ss"
+)
+
+type options struct {
+	clusterAmount int
+	outputFile    string
+
+	f *factory.Factory
+}
+
+func NewSnapshotCommand(f *factory.Factory) *cobra.Command {
+	opts := options{
+		f: f,
+	}
+
+	cmd := &cobra.Command{
+		Short:   "Takes a snapshot of deployments in oldest and ready clusters.",
+		Use:     CommandName,
+		Aliases: []string{CommandAlias},
+		Args:    cobra.NoArgs,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if opts.clusterAmount < 1 || opts.clusterAmount > 30 {
+				return errors.New("clusterAmount must be between 1 and 30")
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return run(&opts)
+		},
+	}
+
+	cmdutil.AddClusterAmount(cmd, &opts.clusterAmount)
+	cmdutil.AddOutputFile(cmd, &opts.outputFile)
+
+	return cmd
+}
+
+func run(opts *options) error {
```

Comment:
Why a pointer here? I think implementing this function by value achieve the same goal

Reply:
Same as above

Reply:
same as above :)

Reply:
Haha ... I've noticed from kas they tend to use pointers everywhere. Imho pointers are a tool in a wide toolset and should not be used just because it is available. The default approach should be to use value implementation instead of reference.

I know it is a copy and paste issue and I am bringing this up to discussion because I saw it earlier this week, sorry for using your PR as a platform :-)